### PR TITLE
Pmd7 upgrade 7.2.0 sql set 2

### PR DESCRIPTION
**Pull request by jborgers**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -110,10 +110,16 @@
             <version>3.2.0</version>
         </dependency>
         <!-- https://mvnrepository.com/artifact/org.hibernate.orm/hibernate-core -->
-        <dependency>
+        <!--dependency>
             <groupId>org.hibernate.orm</groupId>
             <artifactId>hibernate-core</artifactId>
             <version>6.6.1.Final</version>
```
Comment (stokpop) [inline pom.xml:116]: different hibernate versions orm and without orm?

File: src/main/resources/category/java/sql.xml [hunk 2]
```diff
@@ -33,90 +33,94 @@
...
                     <![CDATA[
-//MethodDeclaration/Block//PrimaryExpression/PrimaryPrefix/Name[ends-with(@Image, '.getSingleResult')]
-[ancestor::MethodDeclaration[count(.//VariableInitializer/Expression/PrimaryExpression/PrimarySuffix[@Image='createQuery']) = 1
-and
-count(.//PrimaryExpression/PrimaryPrefix/Name[ends-with(@Image, '.getResultList')]) +
-count(.//PrimaryExpression/PrimaryPrefix/Name[ends-with(@Image, '.getSingleResult')])
-> 1]]
+//MethodCall[pmd-java:matchesSig('javax.persistence.Query#getSingleResult()') or pmd-java:matchesSig('jakarta.persistence.Query#getSingleResult()')][
+    ancestor::MethodDeclaration[
+        count(.//MethodCall[@MethodName='createQuery']) = 1
```
Comment (stokpop) [inline src/main/resources/category/java/sql.xml:53]: no matchesSig for the count methods?

File: src/main/resources/category/java/sql.xml [hunk 3]
```diff
@@ -33,90 +33,94 @@
...
+                <value>
+                    <![CDATA[
+//MethodCall[(
+        @MethodName='in' and (
+            ./TypeExpression[pmd-java:typeIs('org.hibernate.criterion.Restrictions')]
+            or ./VariableAccess[pmd-java:typeIs('javax.persistence.criteria.CriteriaBuilder') or pmd-java:typeIs('jakarta.persistence.criteria.CriteriaBuilder')]
+        )
+    )
+    or pmd-java:matchesSig('javax.persistence.criteria.Expression#in(_*)')
+    (:or pmd-java:matchesSig('javax.persistence.criteria.CriteriaBuilder.<T>#in(_*)') -how to match <T>? :)
```
Comment (stokpop) [inline src/main/resources/category/java/sql.xml:85]: does not match without the `<T>`?

**Review by stokpop — APPROVED** [review]
Some questions, otherwise fine.

File: pom.xml (hunk 1 above)
Comment (jborgers) [reply pom.xml:116]: 6 commented out

File: src/main/resources/category/java/sql.xml (hunk 2 above)
Comment (jborgers) [reply src/main/resources/category/java/sql.xml:53]: shorter, type checked

File: src/main/resources/category/java/sql.xml (hunk 3 above)
Comment (jborgers) [reply src/main/resources/category/java/sql.xml:85]: will remove

**Pull request merged**